apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/codegen/java/selections/CompiledSelectionsBuilder.kt

```diff
@@ -68,7 +68,8 @@ class CompiledSelectionsBuilder(
   }
 
   private fun List<GQLSelection>.walk(name: String, private: Boolean, parentType: String): List<FieldSpec> {
-    val propertyName = resolveNameClashes(usedNames, context.layout.propertyName(name))
+    val maybeEscapedName = if (private) context.layout.escapedPropertyName(name) else context.layout.propertyName(name)
+    val propertyName = resolveNameClashes(usedNames, maybeEscapedName)
```

Comment:
Looks like the only case where this is public is the `root` field? In that case,  you could rename the root field to `__root` (technically source breaking but I think that's ok?) and escape indiscriminately, just like for variables?

All in all, I think I'd do this:

* __typename -> never escape
* __root -> never escape
* variables -> always escape
* add a new `compiledSelectionsName()` that would always escape.

Reply:
Digging into this again. Do you remember why we needed to escape the compiled selections? They look like this:

```kotlin
public object TestQuerySelections {
  private val __onDroid: List<CompiledSelection> = listOf(..)
  private val __hero: List<CompiledSelection> = listOf(..)
  // ...
  public val __root: List<CompiledSelection> = listOf(..)
}
```

This is all private (except __root) and this file has some code to resolve name clashes: https://github.com/apollographql/apollo-kotlin/blob/89a439cbaa1a9b6bc02427df7f2d13abf8e953f3/apollo-compiler/src/main/kotlin/com/apollographql/apollo3/compiler/codegen/kotlin/selections/CompiledSelectionsBuilder.kt#L51. So escaping doesn't sound needed? Do you remember the details?

Reply:
You're right, it looks like escaping is unnecessary - not sure why we did this 😅 Maybe we overlooked the fact that only `__root` is public?